epics/pv.py

```diff
@@ -355,13 +353,13 @@ def _set_charval(self, val, call_ca=True):
             return val
         # char waveform as string
         if ntype == dbr.CHAR and self.count < ca.AUTOMONITOR_MAXLENGTH:
-            if isinstance(val, ca.numpy.ndarray):
+            if ca.HAS_NUMPY and isinstance(val, ca.numpy.ndarray):
```

Comment:
what if one doesn't have NUMPY installed?   Doesn't this test now FAIL HARD?
What does this have to do with the issue?